### ISSUE TITLE
Add event handlers through chaining (much like validators)

### ISSUE THREAD

**Issue by janpieterz** [formstate/formstate#17]
I propose the following API addition (one for each handler, see #14 for handler additions):

```typescript
onChangeAfterValidation: (...handlers: (() => void)[]) => this;
```
This would allow the following configuration:
```typescript
public repeated= new FieldState<boolean>({
        value: false
    })
    .validators(val => val !== false && '')
    .onChangeAfterValidation(() => {console.debug('handler1')}, () =>{console.debug('handler2'});
```

**Comment by basarat** [comment]
> .onChangeAfterValidation(() => {console.debug('handler1')}, () =>{console.debug('handler2'});

* I tried to add `onChange` but its difficult as an API user to be exactly clear what order the function should be called (e.g. https://github.com/formstate/formstate/blob/adb9e6138f5c1a43bfb2f051042df939f1b6ebec/src/core/fieldState.ts#L96 before or after `$` and `error` are changed ... should it even be called on reinitValue and questions like that). It resulted in edge cases that were difficult to predict so left it out completely. That said `onChangeAfterValidation` is what `setCompositionParent` gives https://github.com/formstate/formstate/blob/adb9e6138f5c1a43bfb2f051042df939f1b6ebec/src/core/fieldState.ts#L189 We can make it return `this` so its more fluent if you want :rose:

* If the functions do not depend on each other (e.g. validators do depend on previous validations succeeding) would rather take a single function that you can compose into a bigger function yourself e.g. `.onUpdate(() => {console.debug('handler1'); console.debug('handler2')});`

**Comment by janpieterz** [comment]
So this one was more focusing on the simple chaining of configuration (much like the `validators` ) or the `onUpdate` etc from #16 . It seems that chaining the configuration is in general what direction you'd like to go in there.

I'm a little unclear on your first comment. Would you be ok, regardless of the extra hooks from #14, if the hooks that are present on the `FieldState` are configurable through chaining? I'll start a PR for #16 today, will take this on right away if so!

You're right about just making one handler btw, that's a nicer and less confusing API.

**Comment by basarat** [comment]
Released with `0.18.0` :rose: Thanks @janpieterz :heart:
